# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceWorker.h
# Commit: 46e9f06636da
# Full Hash: 46e9f06636da219144df99dc0f0cb96425eb08cb
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-12-08 03:42:13
# Regressor Bug: 1744025
# File Overlap Count: 1
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in PerformanceWorker. r=dom-worker-reviewers,edenchuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132799
# ==============================================================================

diff -r 913cdee4824f -r 46e9f06636da dom/performance/PerformanceWorker.h
--- a/dom/performance/PerformanceWorker.h	Tue Dec 07 16:47:17 2021 +0000
+++ b/dom/performance/PerformanceWorker.h	Tue Dec 07 16:47:17 2021 +0000
@@ -8,12 +8,11 @@
 #define mozilla_dom_PerformanceWorker_h
 
 #include "Performance.h"
+#include "mozilla/dom/WorkerPrivate.h"
 
 namespace mozilla {
 namespace dom {
 
-class WorkerPrivate;
-
 class PerformanceWorker final : public Performance {
  public:
   explicit PerformanceWorker(WorkerPrivate* aWorkerPrivate);
@@ -94,7 +93,7 @@
   }
 
  private:
-  WorkerPrivate* mWorkerPrivate;
+  CheckedUnsafePtr<WorkerPrivate> mWorkerPrivate;
 };
 
 }  // namespace dom
